File: app/src/main/java/igrek/songbook/info/errorcheck/UIErrorHandler.java
package igrek.songbook.info.errorcheck;


import javax.inject.Inject;

import igrek.songbook.dagger.DaggerIoc;
import igrek.songbook.info.logger.Logger;
import igrek.songbook.info.logger.LoggerFactory;
import igrek.songbook.info.UiInfoService;

public class UIErrorHandler {
	
	@Inject
	UiInfoService uiInfoService;
	
	private Logger logger = LoggerFactory.getLogger();
	
	private UIErrorHandler() {
	}
	
	public static void showError(Throwable t) {
		new UIErrorHandler()._handleError(t);
	}
	
	private void _handleError(Throwable t) {
		DaggerIoc.getFactoryComponent().inject(this);
		logger.error(t);
		uiInfoService.showInfo("Error occurred: " + t.getMessage());
	}
	
}

File: app/src/main/java/igrek/songbook/info/errorcheck/UIErrorHandler.kt
package igrek.songbook.info.errorcheck


import igrek.songbook.R
import igrek.songbook.info.UiInfoService
import igrek.songbook.info.UiResourceService
import igrek.songbook.info.logger.LoggerFactory
import igrek.songbook.inject.LazyExtractor
import igrek.songbook.inject.LazyInject
import igrek.songbook.inject.appFactory


class UIErrorHandler private constructor(
        uiInfoService: LazyInject<UiInfoService> = appFactory.uiInfoService,
        uiResourceService: LazyInject<UiResourceService> = appFactory.uiResourceService,
) {
    private val uiInfoService by LazyExtractor(uiInfoService)
    private val uiResourceService by LazyExtractor(uiResourceService)

    private val logger = LoggerFactory.logger

    private fun handleError(t: Throwable) {
        logger.error(t)
        uiInfoService.showInfo(uiResourceService.resString(R.string.error_occurred_s, t.message))
    }

    companion object {
        fun showError(t: Throwable) {
            UIErrorHandler().handleError(t)
        }
    }

}
